scene/gui/file_dialog.cpp

```diff
@@ -702,7 +788,7 @@ void FileDialog::set_mode(Mode p_mode) {
 	if (mode == MODE_OPEN_FILES) {
 		tree->set_select_mode(Tree::SELECT_MULTI);
 	} else {
-		tree->set_select_mode(Tree::SELECT_SINGLE);
+		tree->set_select_mode(Tree::SELECT_ROW);
```

Comment:
Due to this Line, When running tree->deselect_all() in line 346 the item->deselect(selected_col) in tree.cpp line 3210 fails to deselect since it is not among the Modes mentioned in tree.cpp line 3190, i am not sure why that is, still figuring out why the if clause in tree.cpp line 3190 makes sense, but that is how you end up with the `ERROR: deselect_all: Condition "item == prev_item" is true.`

Reply:
i checked, returning it to SELECT_SINGLE fixes the issue, not sure why SELECT_ROW was needed, you are still selecting a single file but maybe it has some other effects, i have no idea :D

Reply:
> i checked, returning it to SELECT_SINGLE fixes the issue, not sure why SELECT_ROW was needed, you are still selecting a single file but maybe it has some other effects, i have no idea :D
Because if you do this is what you get
![Screenshot from 2020-06-18 00-26-29](https://user-images.githubusercontent.com/17744127/84952267-6ae61e80-b0fa-11ea-8593-6b92864e351d.png)